packages/vm/lib/evm/opcodes.ts

```diff
@@ -284,13 +280,22 @@ function getFullname(code: number, name: string): string {
  * @returns {OpcodeList} Opcodes dictionary object.
  */
 export function getOpcodesForHF(common: Common): OpcodeList {
-  let opcodeBuilder = { ...opcodes }
+  let opcodeBuilder: any = { ...opcodes }
 
   for (let fork = 0; fork < hardforkOpcodes.length; fork++) {
     if (common.gteHardfork(hardforkOpcodes[fork].hardforkName)) {
       opcodeBuilder = { ...opcodeBuilder, ...hardforkOpcodes[fork].opcodes }
     }
   }
 
-  return opcodeBuilder
+  for (let key in opcodeBuilder) {
+    let baseFee = common.param('gasPrices', opcodeBuilder[key].name.toLowerCase())
+    // explicitly verify that we have defined a base fee
+    if (baseFee === undefined) {
+      throw new Error('base fee not defined for: ' + opcodeBuilder[key].name)
+    }
+    opcodeBuilder[key].fee = common.param('gasPrices', opcodeBuilder[key].name.toLowerCase())
```

Comment:
`opcodeBuilder[key].fee = baseFee`

also it'd be great if someone could have a look at optimizing `common.param` (see #775)